awx/main/scheduler/task_manager.py

```diff
@@ -532,9 +543,22 @@ def process_pending_tasks(self, pending_tasks):
                 ) or InstanceGroup.find_largest_idle_instance(self.graph[rampart_group.name]['instances'], capacity_type=task.capacity_type)
 
                 if execution_instance or rampart_group.is_container_group:
```

Comment:
If it's a container group, then this code is unreachable, so delete `or rampart_group.is_container_group`